bpf/include/bpf/helpers.h

```diff
@@ -84,4 +84,12 @@ static int BPF_FUNC(sock_hash_update, struct bpf_sock_ops *skops, void *map,
 static int BPF_FUNC(msg_redirect_hash, struct sk_msg_md *md, void *map,
 		    void *key, __u64 flags);
 
+/* Socket lookup helpers */
+static struct bpf_sock *BPF_FUNC(sk_lookup_tcp, void *ctx,
+				 struct bpf_sock_tuple *tuple, __u32 tuple_size,
+				 __u64 netns, __u64 flags);
+static struct bpf_sock *BPF_FUNC(sk_lookup_udp, void *ctx,
```

Comment:
In the skb variants, we are using the actual context type as the second parameter (`struct __sk_buff *`), does it make sense to do that here with `bpf_sock_addr *`? This would allow compile-time type checks on the parameters I think?

Reply:
It's not such a big deal, partly the background of this question is my misunderstanding on introducing additional variations of the helper for different contexts rather than just sharing common declarations. I'll mark this as resolved.